telemetry/internal/appinsights/client.go

```diff
@@ -108,6 +108,14 @@ func (c *Client) TrackEvent(name string) {
 	c.NewEvent(name).Inc()
 }
 
+func (c *Client) TrackEventWithProperties(name string, properties map[string]string) {
```

Comment:
There is no need to add a new method, just add the `properties` parameter the `TrackEvent`. That shouldn't have any performance impact.

Also, in the Go toolchain I'm using a nil Client to denote that uploading is disabled. Note that `Event.Add` does nothing if `Event.client` is nil. This case is not handled here. You can easily fix it by moving the client nil check short-circuit from `Event.Add` to `Client.track`.

Reply:
Will do, this was just to avoid touching the existing API for now--there are a decent number more places that need to be touched to pass this through the `Event` type.

Ah, didn't notice the nil receivers.